projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewService.java

```diff
@@ -156,50 +234,34 @@ private FacetDTO getFacetInfo(View view, View.Column column) {
         return new FacetDTO(view.name + "_" + column.name, column.title, column.type, values, booleanValue, min, max);
     }
 
-    public List<FacetDTO> getFacets() {
-        return calculateRead(ds, () -> viewsConfig.views
-                .stream()
-                .flatMap(view ->
-                        view.columns.stream()
-                                .map(column -> getFacetInfo(view, column))
-                                .filter(f -> (
-                                        f.getMin() != null || f.getMax() != null
-                                                || (f.getValues() != null && f.getValues().size() > 1)
-                                                || f.getBooleanValue() != null
-                                ))
-                )
-                .collect(toList()));
+    private LoadingCache<Boolean, List<FacetDTO>> buildFacetCache() {
+        return CacheBuilder.newBuilder()
+                .refreshAfterWrite(1, TimeUnit.DAYS)
```

Comment:
I think it would be better to have more control over when the cache is being rebuild (even though it may not have a huge hit on performance), I would still suggest making the auto-refresh configurable: if, when and how often it happens.

Reply:
to be done today later, probably in the scope of another pr